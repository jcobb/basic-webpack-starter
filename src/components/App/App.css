.appHasMounted {
    background: green;
}

.appNotMounted {
  background: red;
}
