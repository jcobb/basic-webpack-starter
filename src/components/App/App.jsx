import React, { Component, PropTypes } from 'react';
import { Provider } from 'react-redux';
import styles from './App.css';

class App extends Component {
  componentDidMount() {
    this.props.setAppIdle();
  }

  render() {
    const className = this.props.appState.appIdle
      ? styles.appHasMounted
      : styles.appNotMounted;

    return (
      <Provider store={this.props.store}>
        <div className={className}>
          <h2>Hello, world</h2>
        </div>
      </Provider>
    );
  }
}

App.propTypes = {
  // methods
  setAppIdle: PropTypes.func.isRequired,

  // props
  store: PropTypes.any.isRequired,
  appState: PropTypes.shape({
    appIdle: PropTypes.boolean,
  }).isRequired,
};

export default App;
