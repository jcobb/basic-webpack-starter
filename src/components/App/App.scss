.appHasMounted {
  background: white;
}

.appNotMounted {
  background: red;
}
