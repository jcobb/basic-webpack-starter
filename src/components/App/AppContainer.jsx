import { connect } from 'react-redux';
import App from './App';
import {
    setAppIdle,
    openModal,
} from '../../store/actions';

const mapStateToProps = state => ({
    appState: state.appState,
});

const AppContainer = connect(
    mapStateToProps,
    {
        setAppIdle,
        openModal,
    },
)(App);


export default AppContainer;
