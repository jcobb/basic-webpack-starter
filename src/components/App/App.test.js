import React from 'react';
import reduxMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import App from './App';

const mockStore = reduxMockStore([]);
const store = mockStore();
const setAppIdle = jest.fn();

const props = {
  setAppIdle,
  store,
  appState: {
    appIdle: false
  },
};

describe('App',  ()  => {
  beforeEach(() => {
    setAppIdle.mockClear();
  });

  it('should render correctly', () => {
    const component = renderer.create(<App {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should setAppIdle on mount', () => {
    const component = mount(<App {...props} />);

    expect(setAppIdle).toHaveBeenCalledTimes(1);
  });

  it('should render className appHasMounted when appIdle', () => {
    const newProps = {
      ...props,
      appState: {
        appIdle: true,
      },
    };

    const component = shallow(<App {...newProps} />);
    const wrapper = component.find('div');

    expect(wrapper.hasClass('appHasMounted')).toEqual(true);
  })
});
