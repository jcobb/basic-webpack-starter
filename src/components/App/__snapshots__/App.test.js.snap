// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="appNotMounted"
>
  <h2>
    Hello, world
  </h2>
</div>
`;
